src/test/java/org/itsallcode/io/TestCapturingOuputStream.java

```diff
@@ -55,4 +70,39 @@ void testWithBufferedOutputStreamOnTopOfSystemOut() throws IOException
             System.setOut(previousOut);
         }
     }
+
+    @Test
+    void testOutputForwardedWhenCapturing() throws IOException
+    {
+        final OutputStream delegate = mock(OutputStream.class);
+        final CapturingOutputStream stream = new CapturingOutputStream(delegate);
+        stream.capture();
+        final byte[] content = "content".getBytes();
+        stream.write(content);
+        stream.close();
+        verify(delegate).write(content, 0, 7);
+    }
+
+    @Test
+    void testOutputForwardedWhenNotCapturing() throws IOException
+    {
+        final OutputStream delegate = mock(OutputStream.class);
```

Comment:
Using mocking here is a very white-box-style test for implementation in an external library. On the other hand not doing this with a mock will likely create an infinite loop of trying self-proof. Accepted.